frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -771,18 +771,14 @@ function ReaderFooter:unscheduleFooterAutoRefresh()
 end
 
 function ReaderFooter:shouldBeRepainted()
-    local n = 1
-    local widget = UIManager:getNthTopWidget(n)
-    while widget do
-        if widget.name == "ReaderUI" then
-            return true
-        elseif widget.covers_fullscreen or widget.covers_footer then
-            -- (e.g. the virtual keyboard sets widget_covers_footer == true)
-            return false
-        end
-        n = n + 1
-        widget = UIManager:getNthTopWidget(n)
+    -- Defensive approach to avoid painting over other widgets, since dimen isn't reliable enough to do it right...
+    local top_wg = UIManager:getNthTopWidget() or {}
+    if top_wg.name == "ReaderUI" or top_wg.name == "FrontLightWidget" or top_wg.invisible then
+        -- FrontLightWidget is here to allow reflecting changes immediately in the footer, even if it *can* overlap us (if moved).
+        -- Invisible is here for TrapWidget, especially in the context of the Dimmer plugin.
+        return true
```

Comment:
I think it can happen that below an `.invisible` top widget, there is another UI widget that is not ReaderUI (ie. TrapWidget, maybe when doing a menu search). Dimmer (if that's it you try to catch with .invisible) may not be the only one. And are we sure it is only used over ReaderUI, and not when spending time on a DictLookup result ?

Reply:
The first case would lead to a possibly overlapping refresh; the second to a missed refresh (as DictLookup would be over Dimmer, I'd imagine).

Should I just give the Dimmer's Trap a name and use that instead?

Reply:
I have no real idea about what Dimmer is (except that it _must_ be some fine piece of o-engineering widget :) and how/when it is used.
But logically, if the top widget is .invisible (and you want to survice future usages of .invisible), you should check the below widget (which might be ReaderUI or FrontLightWidget, or something else).
ie. `UIManager:topdown_widgets_iter()` and similar "get top widget" functions could be provided a `skip_invisible=true/false` param to skip them ?

Reply:
We'd be back into potentially weird and brittle heuristics; I'd really like to KISS, even if it means missed refreshes ;).

Reply:
In short, I don't particularly care about invisible in and of itself, it was used here solely as a shortcut to get to Dimmer. If it turns out it can be problematic, I'd rather find a way to get at Dimmer specifically instead: i.e., a name.